server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -687,49 +689,77 @@ public AnalyzedRelation visitWithQuery(WithQuery node, StatementAnalysisContext
     protected AnalyzedRelation visitTable(Table<?> node, StatementAnalysisContext context) {
         QualifiedName tableQualifiedName = node.getName();
+        RelationName relationName = RelationName.of(tableQualifiedName, null);
         SearchPath searchPath = context.sessionSettings().searchPath();
-        AnalyzedRelation relation;
         var relationContext = context.currentRelationContext();
 
         var withQuery = relationContext.parentSources()
-            .getAncestor(RelationName.of(tableQualifiedName, null));
+            .getAncestor(relationName);
         if (withQuery != null) {
-            relation = withQuery;
+            relationContext.addSourceRelation(withQuery);
+            return withQuery;
         } else {
-            TableInfo tableInfo;
+            String schemaName = relationName.schema();
+            String tableName = relationName.name();
+
+            ArrayList<String> newSearchPath = new ArrayList<>();
+            Iterables.addAll(newSearchPath, searchPath);
+            if (schemaName != null) {
+                newSearchPath.remove(schemaName);
+                newSearchPath.addFirst(schemaName);
+            }
+
+            AnalyzedRelation relation = null;
+            for (String schema : newSearchPath) {
+                RelationName relNameWithIteratingSchema = new RelationName(schema, tableName);
+                try {
+                    TableInfo tableInfo = schemas.getTableInfo(relNameWithIteratingSchema);
+                    if (tableInfo instanceof DocTableInfo docTable) {
+                        // Dispatching of doc relations is based on the returned class of the schema information.
+                        relation = new DocTableRelation(docTable);
+                    } else {
+                        relation = new TableRelation(tableInfo);
+                    }
+                    break;
+                } catch (RelationUnknown | SchemaUnknownException e1) {
+                    View view;
+                    try {
+                        view = schemas.getView(relNameWithIteratingSchema);
+                    } catch (RelationUnknown e2) {
+                        continue;
+                    }
+                    ViewMetadata viewMetadata = view.metadata();
+                    Statement viewQuery = SqlParser.createStatement(viewMetadata.stmt());
+                    AnalyzedRelation resolvedView = context.withSearchPath(
+                            viewMetadata.searchPath(),
+                            newContext -> viewQuery.accept(this, newContext)
+                    );
+                    relation = new AnalyzedView(view.name(), viewMetadata.owner(), resolvedView);
+                    break;
+                }
+            }
+            // getTableInfo/getView do not throw the expected exceptions, call 'resolve' variants
+            // which generate the expected exceptions such as SchemaUnknown/RelationUnknown/OperationOnInaccessibleRelation
```

Comment:
Why not raise it manually?

We should probably also check other uses of `resolveTableInfo` since it still uses the search-path, which is kinda broken. Ideally we'd remove that.

Reply:
> We should probably also check other uses of resolveTableInfo since it still uses the search-path, which is kinda broken. Ideally we'd remove that.

Then instead of having the logic change done in `visitTable()` could we replace `Schemas.resolveTableInfo` to something like `Schemas.resolveRelationInfo` similar to `Schemas.resolveRelation` but returns a `TableInfo` or `ViewInfo`.

Reply:
> Why not raise it manually?

Tried to raise it manually in `RelationAnalyzer.visitTable`, but it forces exposing several methods out of `Schemas` and the logic became very similar to `Schemas.resolveTableInfo`. Again, it seems better to move the logic into `Schemas` as `resolveRelationInfo`.

> We should probably also check other uses of resolveTableInfo

Other uses seem fine to me since all of them just need to resolve tables(no following try-catch for resolving reviews).

I think as a follow up, I can replace `resolveTableInfo` with `resolveRelationInfo` and resolving to wrong views will be prevented based on the `operation`. i.e.:
```
cr> create view v1 as select 1;                                                                                                  
CREATE OK, 1 row affected  (0.179 sec)
cr> create view v2 as select 2;                                                                                                  
CREATE OK, 1 row affected  (0.132 sec)

cr> alter cluster swap table v1 to v2;                                                                                           
RelationUnknown[Relation 'v1' unknown]

-- resolveRelationInfo() will resolve to views then check for acceptable operations
cr> alter cluster swap table v1 to v2;                                                                                           
OperationOnInaccessibleRelationException[The relation "doc.v1" doesn't support or allow ALTER operations, as it is read-only.]
```